src/raised-button.jsx

```diff
@@ -15,6 +13,92 @@ function validateLabel(props, propName, componentName) {
   }
 }
 
+function getStyles(props, state) {
+  const {
+    baseTheme,
+    button,
+    raisedButton,
+  } = state.muiTheme;
+
+  const {
+    disabled,
+    icon,
+    labelPosition,
+    primary,
+    secondary,
+  } = props;
+
+  const amount = (primary || secondary) ? 0.4 : 0.08;
+
+  let backgroundColor = raisedButton.color;
+  let labelColor = raisedButton.textColor;
+
+  if (props.disabled) {
+    backgroundColor = props.disabledBackgroundColor || raisedButton.disabledColor;
+    labelColor = props.disabledLabelColor || raisedButton.disabledTextColor;
+  } else if (props.primary) {
+    backgroundColor = raisedButton.primaryColor;
+    labelColor = raisedButton.primaryTextColor;
+  } else if (props.secondary) {
+    backgroundColor = raisedButton.secondaryColor;
+    labelColor = raisedButton.secondaryTextColor;
+  }
+
+  const styles = {
+    root: {
+      display: 'inline-block',
+      minWidth: props.fullWidth ? '100%' : button.minWidth,
+      height: button.height,
+      transition: Transitions.easeOut(),
+    },
+    container: {
+      position: 'relative',
+      height: '100%',
+      width: '100%',
+      padding: 0,
+      overflow: 'hidden',
+      borderRadius: 2,
+      transition: Transitions.easeOut(),
+      backgroundColor: backgroundColor,
+    },
+    label: {
+      position: 'relative',
+      opacity: 1,
+      fontSize: '14px',
+      letterSpacing: 0,
+      textTransform: raisedButton.textTransform || button.textTransform || 'uppercase',
+      fontWeight: Typography.fontWeightMedium,
+      margin: 0,
+      userSelect: 'none',
+      paddingLeft: baseTheme.spacing.desktopGutterLess,
+      paddingRight: baseTheme.spacing.desktopGutterLess,
+      lineHeight: props.style && props.style.height || `${button.height}px`,
+      color: labelColor,
+    },
+    overlay: {
+      backgroundColor: state.hovered && !disabled && ColorManipulator.fade(labelColor, amount),
+      transition: Transitions.easeOut(),
+      top: 0,
+    },
+    overlayWhenHovered: {
+    },
+    ripple: {
+      color: labelColor,
+      opacity: !(primary || secondary) ? 0.1 : 0.16,
+    },
+  };
+
+  if (icon) {
```

Comment:
i think you can squeeze these into the returned object too. with ternaries :laughing: and return it as anonymous object.